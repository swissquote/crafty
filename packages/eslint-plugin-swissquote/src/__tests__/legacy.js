/* global describe, it, expect */

const { prepareCLIEngine, lint } = require("../../test_utils");
const engine = prepareCLIEngine("legacy");

it("Doesn't work with ES6", () => {
  const result = lint(
    engine,
    `
const something = [];

something.push("a value");
`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toBe(0);
  expect(result.errorCount).toBe(1);
});

describe("ES5 formatting", () => {
  it("Gives no warning on correct code", () => {
    const result = lint(
      engine,
      `
/* global jQuery */
(function($) {
  "use strict";

  function getjQueryObject(string) {
    // Make string a vaild jQuery thing
    var jqObj = $("");
    try {
      jqObj = $(string);
    } catch (e) {
      jqObj = $("<span />").html(string);
    }
    return jqObj;
  }

  $.print = $.fn.print = function(newOptions) {
    var defaults = {
      globalStyles: true,
      mediaPrint: false,
      someObject: "actuallyNot"
    };

    // Print a given set of elements
    var options = $.extend({}, defaults, newOptions || {});
    var $styles = getjQueryObject(options.someObject);
    if (options.globalStyles) {
      // Apply the stlyes from the current sheet to the printed page
      $styles = $("style, link, meta, base, title");
    } else if (options.mediaPrint) {
      // Apply the media-print stylesheet
      $styles = $("link[media=print]");
    }
    return $styles;
  };
})(jQuery);
`
    );

    expect(result.messages).toMatchSnapshot();
    expect(result.warningCount).toBe(0);
    expect(result.errorCount).toBe(0);
  });

  it("Warns on wrong format", () => {
    const result = lint(
      engine,
      `

function test() { "use strict"; fetch("This is spartaaa"); }

test()

`
    );

    expect(result.messages).toMatchSnapshot();
    expect(result.warningCount).toBe(0);
    expect(result.errorCount).toBe(2);
  });
});
