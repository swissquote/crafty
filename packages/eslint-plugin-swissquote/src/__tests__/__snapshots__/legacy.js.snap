// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doesn't work with ES6 1`] = `
Array [
  Object {
    "column": 1,
    "fatal": true,
    "line": 1,
    "message": "Parsing error: The keyword 'const' is reserved",
    "ruleId": null,
    "severity": 2,
  },
]
`;

exports[`ES5 formatting Gives no warning on correct code 1`] = `Array []`;

exports[`ES5 formatting Warns on wrong format 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 60,
    "endLine": 2,
    "fix": Object {
      "range": Array [
        0,
        60,
      ],
      "text": "function test() {
  \\"use strict\\";
  fetch(\\"This is spartaaa\\");
",
    },
    "line": 1,
    "message": "Replace \`⏎function·test()·{·\\"use·strict\\";·fetch(\\"This·is·spartaaa\\");·\` with \`function·test()·{⏎··\\"use·strict\\";⏎··fetch(\\"This·is·spartaaa\\");⏎\`",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
  Object {
    "column": 7,
    "endColumn": 1,
    "endLine": 5,
    "fix": Object {
      "range": Array [
        69,
        70,
      ],
      "text": ";",
    },
    "line": 4,
    "message": "Replace \`⏎\` with \`;\`",
    "nodeType": null,
    "ruleId": "@swissquote/swissquote/prettier/prettier",
    "severity": 2,
  },
]
`;
